bundler/spec/commands/remove_spec.rb

```diff
@@ -616,4 +616,43 @@
       expect(out).to include("foo could not be removed.")
     end
   end
+
+  describe "with comments that mention gems" do
+    context "when comment is a separate line comment" do
+      it "does not remove the line comment" do
+        gemfile <<-G
+          source "#{file_uri_for(gem_repo1)}"
+
+          # gem "rack" might be used in the future
+          gem "rack"
+        G
+
+        bundle "remove rack"
+
+        expect(out).to include("rack was removed.")
+        gemfile_should_be <<-G
+          source "#{file_uri_for(gem_repo1)}"
+
+          # gem "rack" might be used in the future
+        G
+      end
+    end
+
+    context "when gem specified for removal has an inline comment" do
```

Comment:
How about a situation where `gem "rack"` is an inline comment to a different gem, like

```
gem "puma" # implements interface provided by gem "rack"
```

Does it work?

Reply:
Thank you for the feedback. I included two new tests to show this situation is now covered. I hope this is what you had in mind. Please let me know if I can make any improvements.